core/include/assert.h

```diff
@@ -109,6 +109,13 @@ NORETURN void _assert_failure(const char *file, unsigned line);
 #define assert(cond) ((cond) ? (void)0 : core_panic(PANIC_ASSERT_FAIL, assert_crash_message))
 #endif
 
+#if __STDC_VERSION__ >= 201112L && !defined __cplusplus
+/**
+ * @brief define c11 static_assert() macro
+ */
+#define static_assert _Static_assert
```

Comment:
Wouldn't it make more sense to define
```
#if __STDC_VERSION__ >= 201112L && !defined __cplusplus
#define static_assert(...) _Static_assert(__VA_ARGS__)
#else
#define static_assert(...)
#endif
```

... so code compiles for both >=c11 and <c11 compilers?

Reply:
what do you think of 
```
#if __STDC_VERSION__ >= 201112L && !defined __cplusplus
/**
 * @brief define c11 static_assert() macro
 */
#define static_assert _Static_assert
#else
#define static_assert(test, msg) assert(test)
#endif
```
so static asserts are mapped on normal asserts if you don't compile for c11?

Reply:
I thought static asserts can be used at places where others can't, like at global level:

```
static_assert(sizeof(unsigned long) == 4);
void main() {...}
```

so defining to normal asserts would make that use case impossible.

Reply:
Yes are right, I will switch to your proposal

Reply:
I tested your proposal with
```c
#include <assert.h>
int test;
static_assert(sizeof(test) == 4, "msg");
int main(void)
{
    return 0;
}
```
but when I compile for c99. I get 
```
~/RIOT/tests/static/main.c:25:41: error: ISO C does not allow extra
;’ outside of a function [-Werror=pedantic]
 ```
So it seems that if you want to use this c11 feature you should compile your code with c11.

Reply:
@kaspar030 What do you want me todo?

Reply:
can you try defining the non-c11 case to ```struct _static_assert_dummy```?

that would turn a non-c11 call to "static_assert(...);" to a forward declaration, which can be in a file multiple times and doesn't have to be used.

Reply:
(thanks for the nudge)

Reply:
I tested it and your proposal worked. OK to squash?

Reply:
yes, please squash!